hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/listener/ClientSmartListenerService.java

```diff
@@ -165,7 +165,7 @@ public void run() {
                     try {
                         invoke(registrationKey, member.getAddress());
                     } catch (Exception e) {
-                        logger.warning("Listener " + registrationKey + " can not added to new member " + member);
+                        logger.warning("Listener " + registrationKey + " can not added to new member " + member + ". Reason:" + e);
```

Comment:
Please use logger.warning("somemessage", e).

This way you don't loose the stacktrace.